CMakeLists.txt

```diff
@@ -52,6 +52,7 @@ endif()
 option( WITH_REFERENCE_BACKEND "With Reference backend" ON )
 option( WITH_ALP_REFERENCE_BACKEND "With Reference Dense backend" ON )
 option( WITH_OMP_BACKEND "With OMP backend" ON )
+option( WITH_ALP_OMP_BACKEND "With OMP Dense backend" ON )
```

Comment:
Perhaps one question related to the build & test changes would be, are these changes in line with current [docs](https://github.com/Algebraic-Programming/ALP/blob/master/docs/Build_and_test_infra.md ) or did you have to diverge in any ways? If so we should at least keep track of this for integrating into the official documentation at some point.